RocketCast/XMLParser.swift

```diff
@@ -10,92 +10,128 @@ import Foundation
 
 class XMLParser: NSObject {
     
-    var strXMLData:String = ""
-    var currentElement:String = ""
-    var passData:Bool=false
-    var passName:Bool=false
+    var element = NSString()
     
-    
-    var url: NSURL?
-    var xmlContent:NSDictionary?
-    init(url: NSURL) {
+    var podcast = PodcastModel()
+
+    init(url: String) {
         super.init()
-        self.url = url
-        self.xmlContent = parseData()
-    }
-    
-    func getPodcastAuthor() {
+        if let data = NSData(contentsOfURL:NSURL(string: url)!) {
+            parseData(data)
+        } else {
+            Log.error("There's nothing in the data")
```

Comment:
More information on this error. Something like `data from url:\(url) should not be nil`